packages/protocol/contracts/layer1/core/impl/Inbox.sol

```diff
@@ -193,6 +208,10 @@ contract Inbox is IInbox, ICodec, IForcedInclusionStore, EssentialContract {
             uint48 lastFinalizedProposalId = _coreState.lastFinalizedProposalId;
             require(nextProposalId > 0, ActivationRequired());
 
+            uint256 bondAmount = _livenessBond;
+            // During permisionless proving we can set the bond to 0, avoiding the cost of this check
+            if (bondAmount > 0) LibBonds.debitBond(_bondingStorage, msg.sender, bondAmount);
```

Comment:
One idea I really like is **lazy bond processing**: at proposal time, we only verify that the actor has enough bond locked and that it stays locked long enough. Any penalties can be applied later (i.e., not during proposal submission).

That means proposing only needs a single SLOAD, and we can skip the “debit then refund” liveness bond flow, which should cut gas costs significantly.